test/functional/tests/misc/test_device_capabilities.py

```diff
@@ -0,0 +1,229 @@
+            cache, core, error_output = prepare_cas_device(cache_device, core_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(cache_device, core_device, cache, core, error_output)
+        with TestRun.step("Recreate CAS device with switched cache and core devices."):
+            cache, core, error_output = prepare_cas_device(core_device, cache_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(core_device, cache_device, cache, core, error_output)
+
+    with TestRun.group(f"Standard core device | max_sectors_kb = 128"):
+        core_device.set_max_io_size(Size(128, Unit.KibiByte))
+        with TestRun.step("Start cache and add prepared core device as core."):
+            cache, core, error_output = prepare_cas_device(cache_device, core_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(cache_device, core_device, cache, core, error_output)
+        with TestRun.step("Recreate CAS device with switched cache and core devices."):
+            cache, core, error_output = prepare_cas_device(core_device, cache_device)
+        with TestRun.step("Compare capabilities for CAS device, cache and core "
+                          "(or check proper error if logical sector mismatch occurs)."):
+            compare_capabilities(core_device, cache_device, cache, core, error_output)
+
+
+# Methods used in test
+
+def create_scsi_debug_device(sector_size: int, physblk_exp: int, dev_size_mb=1024):
+    scsi_debug_params = {
+        "delay": "0",
+        "virtual_gb": "200",
+        "dev_size_mb": str(dev_size_mb),
+        "sector_size": str(sector_size),
+        "physblk_exp": str(physblk_exp)
+    }
+    scsi_debug = TestRun.plugin_manager.get_plugin('scsi_debug')
+    scsi_debug.params = scsi_debug_params
+    scsi_debug.reload()
+    return TestRun.scsi_debug_devices[0]
+
+
+def prepare_cas_device(cache_device, core_device):
+    cache = casadm.start_cache(cache_device, force=True)
+    try:
+        cache_dev_bs = disk_utils.get_block_size(cache_device.get_device_id())
+        core_dev_bs = disk_utils.get_block_size(core_device.get_device_id())
+        core = cache.add_core(core_device)
+        if cache_dev_bs > core_dev_bs:
+            TestRun.LOGGER.error(
+                f"CAS device started with cache device logical block size ({cache_dev_bs}) "
+                f"greater than core device logical block size ({core_dev_bs})")
+        return cache, core, None
+    except CmdException as e:
+        if cache_dev_bs <= core_dev_bs:
+            raise
```

Comment:
If you rise exception, wouldn't if fail whole test run (I mean whole pytest instance) instead of failing just single test?